activerecord/lib/active_record/connection_adapters/schema_cache.rb

```diff
@@ -33,10 +33,10 @@ def self.read(filename, &block)
       private_class_method :read
 
       attr_reader :version
-      attr_accessor :connection
+      thread_cattr_accessor :connection
```

Comment:
That means all schema caches used in a thread have the same connection. That won't work with multiple databases.

Reply:
Ah, you're right. I forgot about multiple databases.